apex/contrib/optimizers/distributed_fused_adam.py

```diff
@@ -494,16 +582,51 @@ def __init__(self,
             'no_copy' in inspect.getfullargspec(torch.distributed.gather).args
         )
 
+        # Make sure parameter values are same across processes
+        self._broadcast_params()
+
+        # Lock for callbacks
+        self._lock = threading.Lock()
         # Attach hooks for gradient synchronization
         self._register_post_backward_hooks()
+        # Attach hooks for param synchronization
+        if self.overlap_param_sync:
+            self._register_pre_forward_hooks()
+
+    def _broadcast_params(self):
+        """Broadcast parameter values from root rank"""
+        sync_requests = []
+        process_group = self.process_group
+        process_group._start_coalescing()
```

Comment:
this part could require a similar change as https://github.com/NVIDIA/apex/pull/1551/commits/d42f78118047cb680da0aec77ee1e23124e631ef

Reply:
Done